### DISCUSSION TITLE
[Feature]: Support for "fastly" serverBuildTarget

### DISCUSSION THREAD

**Discussion by harmony7** [remix-run/remix#4613] [Proposals]
At Fastly we'd like to get support for [Fastly Compute@Edge](https://developer.fastly.com/learning/compute/javascript/) added to Remix.

We've done some work and we've been able to build both a runtime (implements the stuff in `@remix-run/server-runtime`) and a server adapter (the part that is called by user code). We've also been able to build a template that can be used to kickstart an app with `remix create`. We'll be releasing them soon under [Fastly Labs](https://www.fastlylabs.com).

Fastly Compute@Edge's JavaScript runtime works using the Service Worker API model, similar to the approach taken by Cloudflare Workers.  For this reason, using a `serverBuildTarget` value of `cloudflare-workers` works for now.  So I'd like to contribute code to add `fastly-compute-js` as one of the values of `serverBuildTarget`: 1) when building an appropriate esbuild config; 2) to get serverBareModulesPlugin to bundle everything.

EDIT: runtime/adapter/template will be maintained by Fastly, we only need to get the serverBuildTarget added.

**Comment by harmony7** [comment]
Related: https://github.com/remix-run/remix/discussions/2888

**Comment by harmony7** [comment]
Have created this PR
https://github.com/remix-run/remix/pull/4860

Once this is accepted, it will be possible to use the adapter we created at
https://github.com/fastly/remix-compute-js

**Comment by MichaelDeBoey** [comment]
#4841 is now merged, which will provide new extra config options (`serverConditions`, `serverMainFields` & `serverMinify`) that can be used to implement this instead

**Comment by brophdawg11** [comment]
This discussion is being closed as part of the Remix v3 repository cleanup.

Remix v1/v2 was [upstreamed into React Router v7](https://remix.run/blog/wake-up-remix), so if this proposal is still relevant, please reopen or continue the discussion in the [React Router](https://github.com/remix-run/react-router) repository.
